src/dataflow/DataflowAnnotatedTypeFactory.java

```diff
@@ -196,10 +198,30 @@ public Void visitMethodInvocation(MethodInvocationTree node, AnnotatedTypeMirror
                         processingEnv);
                 TypeMirror tm = type.getUnderlyingType();
                 typeNamesMap.put(tm.toString(), tm);
+                if (tm.getKind() == TypeKind.ARRAY) {
+                    replaceArrayComponentATM((AnnotatedArrayType) type);
+                }
                 type.replaceAnnotation(dataFlowType);
             }
             return super.visitMethodInvocation(node, type);
         }
+
+        /**
+         * Add the bytecode default Dataflow annotation for component type of the given {@link AnnotatedArrayType}.
+         *
+         *<p> For multi-dimensional array, this method will recursively add bytecode default Dataflow annotation to array's component type.
+         *
+         * @param arrayAtm the given {@link AnnotatedArrayType}, whose component type will be added the bytecode default.
+         */
+        private void replaceArrayComponentATM(AnnotatedArrayType arrayAtm) {
```

Comment:
Hi Prof. Dietl, there is no  sub-class relationship between `DataflowTreeAnnotator` and `DataflowInferenceTreeAnnotator`... Their common parent class is `TreeAnnotator`, and `DataflowInferenceTreeAnnotator` is also a subclass of `InferenceTreeAnnotator`, so it cannot share the common logic through a `protected` method....